utils/auto_dockerfile_update/update_dockerfiles.py

```diff
@@ -0,0 +1,188 @@
+                print("Branch already exits.")
+                continue
+            try:
+                branch = git_repo.create_head(branch_name)
+                branch.checkout()
+
+                update_dockerfile(file, latest_tag_name)
+
+                git_repo.git.add("*")
+                git_repo.git.commit(m=f"Update from {file['tags']} to {latest_tag_name}")
+                git_repo.git.push('--set-upstream', 'origin', branch)
+
+            except GitCommandError as e:
+                print(e)
+            finally:
+                original_branch.checkout()
+            print(f"Updated {file['path']}")
+
+
+def create_dependencies_json(all_docker_files: List[Dict]) -> Dict:
+    """
+    Create a dictionary with all the docker images and dependent images.
+    Args:
+        all_docker_files (List): list of dockerfiles
+    Returns:
+        {'image name': [dependent dockerfile]}
+    """
+    dependency_json = {f"{file['repo']}/{file['image_name']}": [] for file in all_docker_files}
+
+    for file in all_docker_files:
+        dependency_json[f"{file['repo']}/{file['image_name']}"].append(file)
+
+    return dependency_json
+
+
+def batch(iterable, n=1):
+    l = len(iterable)
+    for ndx in range(0, l, n):
+        yield iterable[ndx:min(ndx + n, l)]
+
+
+def update_internal_base_dockerfile(git_repo: Repo) -> None:
+    """
+    Update internal docker images in batches
+    Args:
+        git_repo (Repo): the repository
+
+    Returns:
+        None
+    """
+    docker_files = get_docker_files(internal=True)
+    dependency_json = create_dependencies_json(docker_files)
+    for base_image, dependency_list in dependency_json.items():
+        curr_repo, image_name = base_image.split('/')
+        latest_tag = get_latest_tag(curr_repo, image_name, "")
+        latest_tag_name = latest_tag['name']
+        latest_tag_last_updated = latest_tag['last_updated']
+        outdated_files = [file for file in dependency_list if
+                          is_docker_file_outdated(file, latest_tag_name, latest_tag_last_updated)]
+        for index, batch_slice in enumerate(batch(outdated_files, 25)):
+            original_branch = git_repo.active_branch
+            branch_name = fr"autoupdate/{base_image}_{index}"
+            if branch_name in git_repo.git.branch("--all"):
+                continue
+            try:
+                branch = git_repo.create_head(branch_name)
+                branch.checkout()
+
+                for file in batch_slice:
+                    update_dockerfile(file, latest_tag_name)
+
+                git_repo.git.add("*")
+                git_repo.git.commit(m=f"Update  dependent images of {base_image} {index}")
+                git_repo.git.push('--set-upstream', 'origin', branch)
+            except GitCommandError as e:
+                print(f"Error creating {branch_name}")
+                print(e)
+            finally:
+                original_branch.checkout()
```

Comment:
let's extract this logic into another function, we could then call it from both update_internal_base_dockerfile and update_external_base_dockerfiles